clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -401,9 +554,164 @@ export class CopybookDownloadService {
+            return true;
+        }
+      }
+    } catch (error) {
+      this.outputChannel?.appendLine(getErrorMessage(error));
+      return false;
+    }
+    return false;
+  }
 
   public reenableFailedRequests() {
     this.dsnDownloader?.reenableFailedRequests();
     this.ussDownloader?.reenableFailedRequests();
   }
 }
+async function searchCopybookinProcessorGroups(
+  documentUri: string,
+  copybookName: string,
+  storagePath: string,
+  pgConfigs: (
+    | string
+    | ZoweDatasetConfigModel
+    | ZoweUssConfigModel
+    | EndevorConfigModel
+  )[],
+  e4eDownloader?: CopybookDownloaderForE4E,
+  dsnDownloader?: CopybookDownloaderForDsn,
+  ussDownloader?: CopybookDownloaderForUss,
+): Promise<vscode.Uri | undefined> {
+  let result: vscode.Uri | undefined;
+
+  for (const config of pgConfigs) {
+    let shouldFound = false;
+    let folders: string = "";
+    let extensions: string[] | undefined;
+    if (typeof config === "string") {
+      folders = config;
+      extensions = await SettingsService.getCopybookExtension(documentUri);
+    } else if (ENVIRONMENT in config && e4eDownloader) {
+      const endevorType = DownloadUtil.endevorConfigToType(config);
+      const profile = await e4eDownloader.getProfileInfo(config.profile);
+      if (!profile) continue;
+      const has = await e4eDownloader.hasElement(
+        profile,
+        endevorType,
+        copybookName,
+      );
+      if (!has) continue;
+      folders = CopybookURI.createDatasetPath(
+        CopybookURI.getEnviromentPath(endevorType, profile),
+        endevorType.use_map ? USE_MAP : "",
+        storagePath,
+        E4E_FOLDER,
+      ).fsPath;
+      shouldFound = true;
+      extensions = [""];
+    } else if (DATASET in config) {
+      const has = await dsnDownloader?.hasMember(
+        config.profile ? config.profile : SettingsService.getProfileName()!,
+        config.dataset,
+        copybookName,
+      );
+      if (!has) continue;
+      folders = CopybookURI.createDatasetPath(
+        config.profile ? [config.profile] : [SettingsService.getProfileName()!],
+        config.dataset,
+        storagePath,
+      ).fsPath;
+      shouldFound = true;
+      extensions = [""];
+    } else if (USS in config) {
+      extensions = await SettingsService.getCopybookExtension(documentUri);
+      const has = await ussDownloader?.hasMember(
+        config.profile ? config.profile : SettingsService.getProfileName()!,
+        config.uss,
+        copybookName,
+        extensions,
+      );
+      if (!has) continue;
+      folders = CopybookURI.createDatasetPath(
+        config.profile ? [config.profile] : [SettingsService.getProfileName()!],
+        config.uss,
+        storagePath,
+      ).fsPath;
+      shouldFound = true;
+    }
+
+    result = searchCopybookInExtensionFolder(
+      copybookName,
+      folders ? [folders] : [],
+      extensions,
+      storagePath,
+    );
+    if (typeof config === "string" && !result) continue;
```

Comment:
I don't think this is needed anymore.
```suggestion
    if (typeof config === "string" && !result) continue;
```